### PR TITLE
docs(core, schema): [CTO-11] Add docs for under documented callback URL feature.

### PR DESCRIPTION
**Pull request by bryanhelmig**
Adding some more documentation to cover the `z.generateCallbackUrl()` usage.

### CODE REVIEW

**Comment by bryanhelmig** [comment]
I think I'll need to include something on how to use `performResume` to handle the `cleanedRequest`/`rawRequest`/`outputData` call to complete a callback.

**Comment by bryanhelmig** [comment]
This just improves the docs, mind giving it a review @xavdid?

File: packages/cli/README-source.md [hunk 1]
```diff
@@ -724,6 +724,49 @@ The `z.cursor` object exposes two methods:
...
+### `z.generateCallbackUrl()`
+
+The `z.generateCallbackUrl()` will return a callback URL your app can `POST` to later for handling long running tasks (like transcription or encoding jobs). In the meantime, the Zap and Task will wait for your response and the user will see the Task marked as waiting.
+
+For example, in your `perform` you might do:
+
+```js
+const perform = async (z, bundle) => {
+  // something like this url:
+  // https://zapier.com/hooks/callback/123/deadbeef-dead-dead-dead-deaddeafbeef/abcdef0123456789abcdef0123456789abcdef01/
```
Comment (xavdid) [inline packages/cli/README-source.md:736]: It's probably random, but maybe random up something that doesn't include `dead-dead-dead` 😁

File: packages/cli/README-source.md [hunk 2]
```diff
@@ -724,6 +724,49 @@ The `z.cursor` object exposes two methods:
...
+  const callbackUrl = z.generateCallbackUrl();
+  await z.request({
+    url: 'https://example.com/api/slow-job',
+    method: 'POST',
+    body: {
+      // ... whatever your app needs
```
Comment (xavdid) [inline packages/cli/README-source.md:742]: Officially, we call apps "integrations", so you should replace that here

File: packages/cli/README-source.md [hunk 3]
```diff
@@ -724,6 +724,49 @@ The `z.cursor` object exposes two methods:
...
+      url: callbackUrl,
+    },
+  });
+  return {"hello": "world"};
```
Comment (xavdid) [inline packages/cli/README-source.md:746]: I'd add a comment to clarify that this object is what populates `bundle.outputData` later (right?). I know it's clarified below, but this is the snippet most people will read

File: packages/cli/README-source.md [hunk 4]
```diff
@@ -724,6 +724,49 @@ The `z.cursor` object exposes two methods:
...
+};
+```
+
+And in your own `/api/slow-job` view (or more likely, an async job) you'd make this request to Zapier:
```
Comment (xavdid) [inline packages/cli/README-source.md:750]: I'd clarify with: 

> you'd make this request to Zapier when the long-running job completes:

File: packages/cli/README-source.md [hunk 5]
```diff
@@ -724,6 +724,49 @@ The `z.cursor` object exposes two methods:
...
+
+```http
+POST /hooks/callback/123/deadbeef-dead-dead-dead-deaddeafbeef/abcdef0123456789abcdef0123456789abcdef01/ HTTP/1.1
+Host: zapier.com
+Content-Type: application/json
+
+{"foo":"bar"}
+```
+
+And finally, in a `performResume` to handle the final step (which will receive `bundle.cleanedRequest`, `bundle.rawRequest`, and `bundle.outputData`):
```
Comment (xavdid) [inline packages/cli/README-source.md:760]: I'd clarify which `bundle.X` properties come from where. It looks like

* `bundle.cleanedRequest` is `{foo: "bar"}`, the payload for the callback url
* `bundle.rawRequest` is the request object for `bundle.cleanedRequest`
* `bundle.outputData` is the data returned from the initial `perform`

**Review by xavdid — APPROVED** [review]
Nice! Left some minor doc clarity things, but it looks good.

File: packages/cli/README-source.md (hunk 1 above)
Comment (bryanhelmig) [reply packages/cli/README-source.md:736]: 😄 Fair enough!

**Pull request merged**